doc/v5.0/man3/MPI_File_get_view.3
.\" -*- nroff -*-
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" Copyright 2015-2016 Research Organization for Information Science
.\"                     and Technology (RIST). All rights reserved.
.\" Copyright (c) 2020      Google, LLC. All rights reserved.
.\" $COPYRIGHT$
.TH MPI_File_get_view 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_File_get_view\fP \- Returns the process's view of data in the file.

.SH SYNTAX
.ft R
.nf
.SH C Syntax
.nf
#include <mpi.h>
int MPI_File_get_view(MPI_File \fIfh\fP, MPI_Offset \fI*disp\fP,
	MPI_Datatype \fI*etype\fP, MPI_Datatype \fI*filetype\fP,
	char \fI*datarep\fP)

.fi
.SH Fortran Syntax (see FORTRAN 77 NOTES)
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_FILE_GET_VIEW(\fIFH\fP, \fIDISP\fP, \fIETYPE\fP,
	\fIFILETYPE\fP, \fIDATAREP\fP, \fIIERROR\fP)
	INTEGER	\fIFH, ETYPE, FILETYPE, IERROR\fP
	CHARACTER*(*)	\fIDATAREP\fP
	INTEGER(KIND=MPI_OFFSET_KIND)	\fIDISP\fP

.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_File_get_view(\fIfh\fP, \fIdisp\fP, \fIetype\fP, \fIfiletype\fP, \fIdatarep\fP, \fIierror\fP)
	TYPE(MPI_File), INTENT(IN) :: \fIfh\fP
	INTEGER(KIND=MPI_OFFSET_KIND), INTENT(OUT) :: \fIdisp\fP
	TYPE(MPI_Datatype), INTENT(OUT) :: \fIetype\fP, \fIfiletype\fP
	CHARACTER(LEN=*), INTENT(OUT) :: \fIdatarep\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT PARAMETER
.ft R
.TP 1i
fh
File handle (handle).

.SH OUTPUT PARAMETERS
.ft R
.TP 1i
disp
Displacement (integer).
.TP 1i
etype
Elementary data type (handle).
.TP 1i
filetype
File type (handle). See Restrictions, below.
.TP 1i
datarep
Data representation (string).
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
The MPI_File_get_view routine returns the process's view of the data
in the file. The current values of the displacement, etype, and
filetype are returned in
.I disp,
.I etype,
and
.I filetype,
respectively.
.sp
The MPI_File_get_view interface allows the user to pass a data-representation string via the \fIdatarep\fP argument.

.SH FORTRAN 77 NOTES
.ft R
The MPI standard prescribes portable Fortran syntax for
the \fIDISP\fP argument only for Fortran 90.  FORTRAN 77
users may use the non-portable syntax.
.sp
.nf
     INTEGER*MPI_OFFSET_KIND \fIDISP\fP
.fi
.sp
where MPI_OFFSET_KIND is a constant defined in mpif.h
and gives the length of the declared integer in bytes.

.SH ERRORS
Almost all MPI routines return an error value; C routines as the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. For MPI I/O function errors, the default error handler is set to MPI_ERRORS_RETURN. The error handler may be changed with MPI_File_set_errhandler; the predefined error handler MPI_ERRORS_ARE_FATAL may be used to make I/O errors fatal. Note that MPI does not guarantee that an MPI program can continue past an error.



doc/v1.4/man3/MPI_File_get_view.3.php
<?php
$topdir = "../../..";
$title = "MPI_File_get_view(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_FILE_GET_VIEW(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_File_get_view</b> - Returns the process&rsquo;s view of data in the file.

<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>
<br>
<pre>C Syntax
    #include &lt;mpi.h&gt;
    int MPI_File_get_view(MPI_File fh, MPI_Offset *disp,
    <tt> </tt>&nbsp;<tt> </tt>&nbsp;      MPI_Datatype *etype, MPI_Datatype *filetype,
              char *datarep)
</pre>
<h2><a name='sect2' href='#toc2'>Fortran Syntax (see FORTRAN 77 NOTES)</a></h2>
<br>
<pre>    INCLUDE &rsquo;mpif.h&rsquo;
    MPI_FILE_GET_VIEW(FH, DISP, ETYPE,
    <tt> </tt>&nbsp;<tt> </tt>&nbsp;       FILETYPE, DATAREP,  IERROR)
    <tt> </tt>&nbsp;<tt> </tt>&nbsp; INTEGER FH, ETYPE, FILETYPE, IERROR
<tt> </tt>&nbsp;<tt> </tt>&nbsp; CHARACTER*(*) DATAREP
    <tt> </tt>&nbsp;<tt> </tt>&nbsp; INTEGER(KIND=MPI_OFFSET_KIND) DISP
</pre>
<h2><a name='sect3' href='#toc3'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
void MPI::File::Get_view(MPI::Offset&amp; disp,
<tt> </tt>&nbsp;<tt> </tt>&nbsp;MPI::Datatype&amp; etype,
<tt> </tt>&nbsp;<tt> </tt>&nbsp;MPI::Datatype&amp; filetype, char* datarep) const
</pre>
<h2><a name='sect4' href='#toc4'>Input Parameter</a></h2>

<dl>

<dt>fh </dt>
<dd>File handle (handle).
<p> </dd>
</dl>

<h2><a name='sect5' href='#toc5'>Output Parameters</a></h2>

<dl>

<dt>disp   </dt>
<dd>Displacement
(integer). </dd>

<dt>etype  </dt>
<dd>Elementary data type (handle).  </dd>

<dt>filetype </dt>
<dd>File type (handle).
See Restrictions, below. </dd>

<dt>datarep </dt>
<dd>Data representation (string).  </dd>

<dt>IERROR </dt>
<dd>Fortran
only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Description</a></h2>
The MPI_File_get_view routine
returns the process&rsquo;s view of the data in the file. The current values of
the displacement, etype, and filetype are returned in  <i>disp,</i> <i>etype,</i> and
 <i>filetype,</i> respectively. <p>
The MPI_File_get_view interface allows the user
to pass a data-representation string via the <i>datarep</i> argument.
<p>
<h2><a name='sect7' href='#toc7'>Fortran 77
Notes</a></h2>
The MPI standard prescribes portable Fortran syntax for the <i>DISP</i> argument
only for Fortran 90.  FORTRAN 77 users may use the non-portable syntax. <p>
<br>
<pre>     INTEGER*MPI_OFFSET_KIND DISP
</pre><p>
where MPI_OFFSET_KIND is a constant defined in mpif.h and gives the length
of the declared integer in bytes.
<p>
<h2><a name='sect8' href='#toc8'>Errors</a></h2>
Almost all MPI routines return
an error value; C routines as the value of the function and Fortran routines
in the last argument. C++ functions do not return errors. If the default
error handler is set to MPI::ERRORS_THROW_EXCEPTIONS, then on error the
C++ exception mechanism will be used to throw an MPI:Exception object. <p>
Before
the error value is returned, the current MPI error handler is called. For
MPI I/O function errors, the default error handler is set to MPI_ERRORS_RETURN.
The error handler may be changed with <a href="../man3/MPI_File_set_errhandler.3.php">MPI_File_set_errhandler</a>; the predefined
error handler MPI_ERRORS_ARE_FATAL may be used to make I/O errors fatal.
Note that MPI does not guarantee that an MPI program can continue past
an error.
<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>Fortran Syntax (see FORTRAN 77 NOTES)</a></li>
<li><a name='toc3' href='#sect3'>C++ Syntax</a></li>
<li><a name='toc4' href='#sect4'>Input Parameter</a></li>
<li><a name='toc5' href='#sect5'>Output Parameters</a></li>
<li><a name='toc6' href='#sect6'>Description</a></li>
<li><a name='toc7' href='#sect7'>Fortran 77 Notes</a></li>
<li><a name='toc8' href='#sect8'>Errors</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
